lib/chusaku.rb

```diff
@@ -19,14 +21,17 @@ def call(flags = {})
       @flags = flags
       @routes = Chusaku::Routes.call
       @changed_files = []
-      controllers_pattern = @flags[:controllers_pattern] || "app/controllers/**/*_controller.rb"
+      controllers_pattern = @flags[:controllers_pattern] || DEFAULT_CONTROLLERS_PATTERN
+      controllers_paths = Dir.glob(Rails.root.join(controllers_pattern))
+
+      @routes.each do |controller, actions|
+        next unless controller
 
-      Dir.glob(Rails.root.join(controllers_pattern)).each do |path|
-        controller = %r{controllers/(.*)_controller\.rb}.match(path)[1]
-        actions = @routes[controller]
-        next if actions.nil?
+        controller_class = "#{controller.underscore.camelize}Controller".constantize
+        source_path = controller_class.instance_method(actions.keys.first.to_sym).source_location[0]
```

Comment:
Is it possible for any chained value to be `nil` here? I specifically wasn't sure if `source_location` could perhaps return an empty array or something. Wondering if safe navigation should be present.